scalafix-cli/src/main/scala/scalafix/internal/jgit/JGitDiff.scala

```diff
@@ -0,0 +1,98 @@
+package scalafix.internal.jgit
+
+import java.nio.file.Path
+
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.api.errors.GitAPIException
+import org.eclipse.jgit.diff.DiffEntry
+import org.eclipse.jgit.diff.DiffFormatter
+import org.eclipse.jgit.patch.FileHeader
+import org.eclipse.jgit.lib.ObjectReader
+import org.eclipse.jgit.lib.Ref
+import org.eclipse.jgit.lib.Repository
+import org.eclipse.jgit.lib.ObjectId
+import org.eclipse.jgit.lib.ProgressMonitor
+import org.eclipse.jgit.dircache.DirCacheIterator
+import org.eclipse.jgit.revwalk.RevCommit
+import org.eclipse.jgit.revwalk.RevTree
+import org.eclipse.jgit.revwalk.RevWalk
+import org.eclipse.jgit.treewalk.AbstractTreeIterator
+import org.eclipse.jgit.treewalk.CanonicalTreeParser
+import org.eclipse.jgit.treewalk.FileTreeIterator
+import org.eclipse.jgit.util.io.NullOutputStream
+import scala.collection.JavaConverters._
+
+object JGitDiff {
+  def apply(workingDir: Path, baseBranch: String): List[GitDiff] = {
+    val builder = new FileRepositoryBuilder()
+    val repository =
+      builder.readEnvironment().setWorkTree(workingDir.toFile).build()
+
+    val oldTree1 = ref(repository, s"refs/heads/$baseBranch")
+    val newTree1 = new FileTreeIterator(repository)
+
+    val diffs = List.newBuilder[GitDiff]
+
+    def path(relative: String): Path =
+      workingDir.resolve(relative)
+
+    def edits(file: FileHeader): Unit = {
+      val changes = List.newBuilder[GitChange]
+      file.toEditList.asScala.foreach { edit =>
```

Comment:
Could this be turned into a map? Builders are great when they're needed but they should be avoided if they can be replaced with a single map (optionally to iterator before and to list after)